### PR TITLE
[T3A3][W11-A1]Tan Wang Leng

### PR DESCRIPTION
**Pull request by yamgent**

### CODE REVIEW

**Comment by yamgent** [comment]
Ready for review.

File: src/seedu/addressbook/data/person/Address.java [hunk 1]
```diff
@@ -1,18 +1,37 @@
...
+    public static final String EXAMPLE = getHumanReadableAddress(Block.EXAMPLE, Street.EXAMPLE, Unit.EXAMPLE,
+            PostalCode.EXAMPLE);
+
+    public static final String MESSAGE_ADDRESS_CONSTRAINTS = "Person address should consist of 4 parts: BLOCK, STREET, UNIT, POSTAL_CODE";
+    public static final String ADDRESS_VALIDATION_REGEX = ".+,.+,.+,.+";
+
+    public final Block block;
+    public final Street street;
+    public final Unit unit;
+    public final PostalCode postalCode;
```
Comment (cmkumar87) [inline src/seedu/addressbook/data/person/Address.java]: Shouldn't you make them private and add accessors to preserve encapsulation?

File: src/seedu/addressbook/data/person/Address.java [hunk 2]
```diff
@@ -25,31 +44,70 @@ public Address(String address, boolean isPrivate) throws IllegalValueException {
...
+     * readable by human (e.g "123, some street, #45-456, 324234")
+     * 
+     * @param block number of the address
+     * @param street name of the address
+     * @param unit number of the address
+     * @param postalCode of the address
+     * @return a string with all the components combined
+     */
+    private static String getHumanReadableAddress(String block, String street, String unit,
+            String postalCode) {
```
Comment (cmkumar87) [inline src/seedu/addressbook/data/person/Address.java]: Why do you need this helper method to be static? Anything declared static inside a class is accessible across instances of the class. Static is not a default qualifier for a class member.

**Comment by cmkumar87** [comment]
@yamgent Good job, in general. 

Please review the comments and close PR. We are now advised to follow this protocol for PR closing to ensure student read our comments.

**Comment by yamgent** [comment]
Thank you!

**Pull request closed without merging**